src/main/java/co/cask/hydrator/plugin/RecordSplitter.java

```diff
@@ -0,0 +1,138 @@
+  private final Config config;
+
+  // Output Schema associated with transform output.
+  private Schema outSchema;
+
+  // Required only for testing.
+  public RecordSplitter(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    config.validate();
+    // Just checking that the operator is valid
+    try {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(Schema.parseJson(config.schema));
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Unable to parse output schema.", e);
+    }
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    try {
+      outSchema = Schema.parseJson(config.schema);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.", e);
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) throws Exception {
+    List<Schema.Field> fields = in.getSchema().getFields();
+    Object valueToSplit = in.get(config.fieldToSplit);
+    if (valueToSplit != null) {
+      String[] records = String.valueOf(valueToSplit).split(config.delimiter);
+      for (String record : records) {
+        StructuredRecord.Builder builder = StructuredRecord.builder(outSchema);
+        for (Schema.Field field : fields) {
+          String name = field.getName();
+          if (outSchema.getField(name) != null && !name.equals(config.fieldToSplit)) {
+            builder.set(name, in.get(name));
+          }
+        }
+        builder.set(config.outputField, record.trim());
+        emitter.emit(builder.build());
+      }
+    }
+  }
+
+  /**
+   * Record Splitter plugin configuration.
+   */
+  public static class Config extends PluginConfig {
+    @Name("fieldToSplit")
+    @Description("Specifies the field to split.")
+    @Macro
+    private final String fieldToSplit;
+
+    @Name("delimiter")
+    @Description("Specifies the delimiter used to split each record. If using escape characters, " +
+                 "be sure to double escape. So \n = \\n")
+    @Macro
+    private final String delimiter;
+
+    @Name("outputField")
+    @Description("Specifies the name of the output field.")
+    @Macro
+    private final String outputField;
+
+    @Name("schema")
+    @Description("Specifies the schema that has to be output.")
+    private final String schema;
+
+    public Config(String fieldToSplit, String delimiter, String outputField, String schema) {
+      this.fieldToSplit = fieldToSplit;
+      this.delimiter = delimiter;
+      this.outputField = outputField;
+      this.schema = schema;
+    }
+
+    public void validate() throws IllegalArgumentException {
+      try {
```

Comment:
Add the validation to check if the input field to split of simple type, The code will fail for complex types